doc/nrf/releases_and_maturity/releases/release-notes-3.1.0.rst

```diff
@@ -0,0 +1,1190 @@
+      Even within the same network, the PDN connection establishment method (PCO vs ePCO) might change when the device operates in NB-IoT or LTE Cat-M1, resulting in missing DNS addresses when one method is used, but not the other.
+      Having a fallback DNS address ensures that the device always has a DNS to fallback to.
+
+  * Updated modem events subscription to persist between functional mode changes.
+
+  * Removed:
+
+    * The deprecated functions ``lte_lc_reduced_mobility_get()``, ``lte_lc_reduced_mobility_set()``, and ``lte_lc_factory_reset()``.
+    * The deprecated macro ``LTE_LC_ON_CFUN()``.
+
+* :ref:`lib_modem_slm` library:
+
+  * Added:
+
+    * The :kconfig:option:`CONFIG_MODEM_SLM_UART_RX_BUF_COUNT` Kconfig option for configuring RX buffer count.
+    * The :kconfig:option:`CONFIG_MODEM_SLM_UART_RX_BUF_SIZE` Kconfig option for configuring RX buffer size.
+    * The :kconfig:option:`CONFIG_MODEM_SLM_UART_TX_BUF_SIZE` Kconfig option for configuring TX buffer size.
+    * The :kconfig:option:`CONFIG_MODEM_SLM_AT_CMD_RESP_MAX_SIZE` Kconfig option for buffering AT command responses.
+
+  * Updated:
+
+      * The software maturity of the library to supported instead of experimental.
+      * The UART implementation between the host device, using the :ref:`lib_modem_slm` library, and the device running the :ref:`Serial LTE Modem <slm_description>` application.
+
+  * Removed:
+
+    * The ``CONFIG_MODEM_SLM_DMA_MAXLEN`` Kconfig option.
+      Use :kconfig:option:`CONFIG_MODEM_SLM_UART_RX_BUF_SIZE` instead.
+    * The ``modem_slm_reset_uart()`` function, as there is no longer a need to reset the UART.
+
+* :ref:`modem_info_readme` library:
+
+  * Added:
+
+    * The :c:func:`modem_info_get_rsrq` function for requesting the RSRQ.
+    * The :c:macro:`SNR_IDX_TO_DB` macro for converting the SNR index to dB.
+
+Multiprotocol Service Layer libraries
+-------------------------------------
+
+* Added an implementation of the API required by the MPSL (defined by :file:`mpsl_hwres.h`) for the nRF53 and nRF54L Series devices.
+
+* Updated the implementation of the following interrupt service routine wrappers:
+
+  * :c:func:`mpsl_timer0_isr_wrapper`
+  * :c:func:`mpsl_rtc0_isr_wrapper`
+  * :c:func:`mpsl_radio_isr_wrapper`
+
+  Now, they do not trigger the kernel scheduler or use any kernel APIs.
+
+  .. note::
+
+     Invoking kernel APIs or triggering the kernel scheduler from Zero Latency Interrupts is considered undefined behavior.
+     Users of MPSL timeslots should not assume that thread rescheduling will occur automatically at the end of a timeslot.
+
+* Fixed an issue where calling the :c:func:`mpsl_lib_uninit` function would prevent calibration of the RC oscillator when MPSL was subsequently re-initialized using the :c:func:`mpsl_lib_init()` function.
+
+  This could happen, for instance, when using bluetooth with the :kconfig:option:`CONFIG_BT_UNINIT_MPSL_ON_DISABLE` Kconfig option enabled.
```

Comment:
```suggestion
  This could happen, for instance, when using Bluetooth with the :kconfig:option:`CONFIG_BT_UNINIT_MPSL_ON_DISABLE` Kconfig option enabled.
```